lib/syskit/network_generation/system_network_generator.rb

```diff
@@ -479,7 +479,10 @@ def self.verify_all_deployments_are_unique(
                 deployment_to_task_map = plan.find_local_tasks(Syskit::TaskContext)
                                              .group_by(&:orocos_name)
 
-                using_same_deployment = deployment_to_task_map.select do |_, tasks|
+                using_same_deployment = deployment_to_task_map.select do |name, tasks|
+                    # There cant be a conflict between tasks that have no deployment
+                    next unless name
+
                     tasks.size > 1
```

Comment:
I would personally prefer just `name && tasks.size > 1`, or doing `deployment_to_task_map.delete(nil)` before the select ...